Source/ThirdParty/libwebrtc/Source/webrtc/sdk/objc/components/video_codec/nalu_rewriter.cc

```diff
@@ -467,6 +468,236 @@ CMVideoFormatDescriptionRef CreateVideoFormatDescription(
+            // max_dec_frame_buffering
+        }
+
+        if (!reader.Ok()) {
+            return { };
+        }
+        return result;
+    }
+
+    static void skipHRDParameters(BitstreamReader& reader)
+    {
+        // cpb_cnt_minus1
+        auto cpb_cnt_minus1 = reader.ReadExponentialGolomb();
+        // bit_rate_scale and bit_rate_scale
+        reader.ConsumeBits(8);
+        for (size_t cptr = 0; cptr < cpb_cnt_minus1; ++cptr) {
+            // bit_rate_value_minus1
+            reader.ReadExponentialGolomb();
+            // cpb_size_value_minus1
+            reader.ReadExponentialGolomb();
+            // cbr_flag
+            reader.ConsumeBits(1);
+        }
+    }
+};
+
+// Table A-1 of H.264 spec
+static size_t maxDpbMbsFromLevelNumber(uint8_t profile_idc, uint8_t level_idc, bool constraint_set3_flag)
+{
+  if ((profile_idc == 66 || profile_idc == 77) && level_idc == 11 && constraint_set3_flag) {
+    // level1b
+    return 396;
+  }
+  H264Level level_casted = static_cast<H264Level>(level_idc);
+
+  switch (level_casted) {
+  case H264Level::kLevel1:
+    return 396;
+  case H264Level::kLevel1_1:
+    return 900;
+  case H264Level::kLevel1_2:
+  case H264Level::kLevel1_3:
+  case H264Level::kLevel2:
+    return 2376;
+  case H264Level::kLevel2_1:
+    return 4752;
+  case H264Level::kLevel2_2:
+  case H264Level::kLevel3:
+    return 8100;
+  case H264Level::kLevel3_1:
+    return 18000;
+  case H264Level::kLevel3_2:
+    return 20480;
+  case H264Level::kLevel4:
+    return 32768;
+  case H264Level::kLevel4_1:
+    return 32768;
+  case H264Level::kLevel4_2:
+    return 34816;
+  case H264Level::kLevel5:
+    return 110400;
+  case H264Level::kLevel5_1:
+    return 184320;
+  case H264Level::kLevel5_2:
+    return 184320;
+  default:
+    RTC_LOG(LS_ERROR) << "Wrong maxDpbMbsFromLevelNumber";
+    return 0;
+  }
+}
+
+uint64_t ComputeH264ReorderSize(const uint8_t* annexb_buffer,
+                                size_t annexb_buffer_size) {
+  AnnexBBufferReader reader(annexb_buffer, annexb_buffer_size);
+  if (!reader.SeekToNextNaluOfType(kSps)) {
+    return 0;
+  }
+  const uint8_t* spsData;
+  size_t spsDataSize;
+
+  if (!reader.ReadNalu(&spsData, &spsDataSize)) {
+    return 0;
+  }
+
+  std::vector<uint8_t> unpacked_buffer = H264::ParseRbsp(spsData, spsDataSize);
+  auto spsAndVui = SpsAndVuiParser::Parse(unpacked_buffer);
+  if (!spsAndVui) {
+    RTC_LOG(LS_ERROR) << "Failed to parse sps.";
+    return 0;
+  }
+
+  if (spsAndVui->pic_order_cnt_type == 2) {
+    return 0;
+  }
+
+  uint64_t max_dpb_mbs = maxDpbMbsFromLevelNumber(spsAndVui->profile_idc, spsAndVui->level_idc, spsAndVui->constraint_set3_flag);
+  uint64_t max_dpb_frames = max_dpb_mbs / ((spsAndVui->width / 16) * (spsAndVui->height / 16));
+  max_dpb_frames = std::min(max_dpb_frames, 16ull);
+
+  if (spsAndVui->bitstream_restriction_flag) {
+    return std::min(spsAndVui->max_num_reorder_frames, max_dpb_frames);
```

Comment:
Per spec
`max_num_ref_frames specifies the maximum number of short-term and long-term reference frames, complementary reference field pairs, and non-paired reference fields that may be used by the decoding     process for inter prediction of any picture in the sequence. max_num_ref_frames also determines the size of the sliding window operation as specified in subclause 8.2.5.3. The value of max_num_ref_frames shall be in the range of 0 to MaxDpbFrames (as specified in subclause A.3.1 or A.3.2), inclusive.)`

max_num_ref_frames defines the sliding window size, which is what defines the max depth of the needed re-ordering queue.

Reply:
According spec, sliding window size is internal to the VTB decoder. max_num_reorder_frames is for the output queue, which has to be  handled by VTB clients themselves. I think we align with the spec by using max_num_reorder_frames, which seems safer.